src/wallet/rpcwallet.cpp

```diff
@@ -2966,7 +2966,7 @@ static UniValue listunspent(const JSONRPCRequest& request)
             std::unique_ptr<SigningProvider> provider = pwallet->GetSolvingProvider(scriptPubKey);
             if (provider) {
                 if (scriptPubKey.IsPayToScriptHash()) {
-                    const CScriptID& hash = CScriptID(boost::get<ScriptHash>(address));
+                    const CScriptID& hash = CScriptID(std::get<ScriptHash>(address));
```

Comment:
Annoyingly, we can't do this while we are still supporting macOS 10.12. Using `std::get` with `std::variant` is not supported on macOS until 10.14, as that is when the libc++ dylib started supporting the `std::bad_variant_access` exception. Thus this PR with `-mmacos-version-min=10.12` fails to compile:
```bash
wallet/rpcwallet.cpp:2969:60: error: 'get<ScriptHash, CNoDestination, PKHash, ScriptHash, WitnessV0ScriptHash, WitnessV0KeyHash, WitnessUnknown>' is unavailable: introduced in macOS 10.14
                    const CScriptID& hash = CScriptID(std::get<ScriptHash>(address));
                                                           ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/variant:1376:16: note: 'get<ScriptHash, CNoDestination, PKHash, ScriptHash, WitnessV0ScriptHash, WitnessV0KeyHash, WitnessUnknown>' has been explicitly marked unavailable here
constexpr _Tp& get(variant<_Types...>& __v) {
```

We could work around this in our code using `std::get_if`, however it's still going to be a problem for 3rd-party dependencies. I've been testing compiling Qt 5.15 LTS, with `-std c++17`, and it also requires macOS 10.14+, as they using get with variant throughout cocoa code.

I'll follow up some discussion in #16684.

Reply:
resolved, now that a52ecc9 is merged